include/bluetooth/mesh/sensor.h

```diff
@@ -208,47 +231,7 @@ struct bt_mesh_sensor_threshold {
 };
 
 /** Sensor channel value format. */
-struct bt_mesh_sensor_format {
-	/** @brief Perform a delta check between two @ref bt_mesh_sensor_value
-	 *         instances.
-	 *
-	 *  @c current and @c previous must have the same format.
-	 *
-	 *  @param[in] current  The current value.
-	 *  @param[in] previous The previous sensor value to compare against.
-	 *  @param[in] delta    The delta to use when checking.
-	 *
-	 *  @return @c true if the difference between @c current and @c previous
-	 *          is bigger than the relevant delta specified in @c delta, @c
-	 *          false otherwise.
-	 */
-	bool (*const delta_check)(const struct bt_mesh_sensor_value *current,
-				  const struct bt_mesh_sensor_value *previous,
-				  const struct bt_mesh_sensor_deltas *delta);
-
-	/** @brief Compare two @ref bt_mesh_sensor_value instances.
-	 *
-	 *  @c op1 and @c op1 must have the same format.
-	 *
-	 *  @param[in] op1 The first value to compare.
-	 *  @param[in] op2 The second value to compare.
-	 *
-	 *  @return 0 if @c op1 == @c op2, 1 if @c op1 > @c op2, -1 otherwise
-	 *          (including if @c op1 and @c op2 are not comparable).
-	 */
-	int (*const compare)(const struct bt_mesh_sensor_value *op1,
-			     const struct bt_mesh_sensor_value *op2);
-
-	/** User data pointer. Used internally by the sensor types. */
-	void *user_data;
-	/** Size of the encoded data in bytes. */
-	size_t size;
-
-#ifdef CONFIG_BT_MESH_SENSOR_LABELS
-	/** Pointer to the unit associated with this format. */
-	const struct bt_mesh_sensor_unit *unit;
-#endif
-};
+struct bt_mesh_sensor_format;
```

Comment:
The idea is that you can't (currently) define your own format, since the message handlers have no way of knowing about these user-defined formats and will just fail to decode the messages. This is specifically why I moved this to the internal header. We can move it back if we add functionality for creating your own formats. I will update the description, though.

Reply:
I think it would make sense to keep this public as someone could already start using their own types.

Reply:
They could use that section. That section consists of `bt_mesh_sensor_type` structs that is public.

Reply:
Hmm. Okay, I'll move it back then. My only concern was that people would start using the callbacks directly in their applications, when they really should use the `bt_mesh_sensor_value_to_...` etc. API functions.